local-proxy/src/server.ts

```diff
@@ -72,17 +73,25 @@ const router = new KoaRouter();
 
 const tmpDir = mkdtempSync(resolvePath(basePath, '..', '.shift_local_proxy_'));
 
-const transpilePromise = babelDir({
-  cliOptions: {
-    filenames: [basePath],
-    outDir: tmpDir,
-  },
-  babelOptions: {
-    sourceMaps: true,
-    plugins: ['@babel/plugin-transform-modules-commonjs'],
-  },
-});
+async function transpileAndCopy() {
+  await babelDir({
+    cliOptions: {
+      filenames: [basePath],
+      outDir: tmpDir,
+    },
+    babelOptions: {
+      sourceMaps: true,
+      plugins: ['@babel/plugin-transform-modules-commonjs'],
+    },
+  });
+  await copy(basePath, tmpDir, {
+    filter(src) {
+      return extname(src) !== '.js';
+    },
+  });
+}
 
+const transpilePromise = transpileAndCopy();
```

Comment:
Not 100% sure, but is this code as safe as it was before? Previously there was only a single promise, so if it fails it can manage its own cleanup. What happens if the `babelDir` succeeds but then the `copy` fails?

Reply:
Cool